### ISSUE TITLE
Add "proprietor" to litter admin page and sort by proprietor/litter name

### ISSUE THREAD

**Issue by cxrodgers** [cxrodgers/MouseColony#13]

**Comment by danieldkato** [comment]
Proprietor could either be 1) a foreign key field of litter itself, or 2) a custom method that traverses breeding cage and returns the proprietor's name as a string.

One thing to note about option 1) is that it allows for the possibility of a litter's proprietor being different from the breeding cage proprietor. I'm not sure whether this is desirable or not. For most litters, the proprietor of the litter should be the same as the proprietor of its breeding cage. In this case, giving litter its own proprietor field would be redundant. 

On the other hand, would it ever be the case that someone transfers a litter to someone else? Would we ever have a situation where someone sets up a breeding cage then eventually transfers a litter (including the mother, per ICM rules) to another user? If so, then we might want to make it possible for a litter's proprietor to be different from the proprietor of the breeding cage the litter came from. In this case, we would definitely want to preserve separate proprietor fields for both breeding cage and litter. 

However, covering this possibility comes at the cost of potentially introducing additional errors; if breeding cage and litter have independent proprietor fields, then it is the user's responsibility to ensure that both of them are accurate. What if someone forgets to set the proprietor of both breeding cage and litter? One way to mitigate such errors would be to write a custom save method for the breeding cage admin that automatically assigns its own proprietor to any newly added litter objects. Then, if anyone ever does transfer a litter, then they can go into the litter admin change page for that litter and manually change the proprietor (which would presumably not happen very often).

Supposing, however, that we would never transfer litters between users, then option 2) precludes the possibility that a litter will erroneously be given a different proprietor from its breeding cage. The problem with this, however, is that since proprietor would not actually be a database field, it's not clear how we could order by proprietor in the list view.

**Comment by danieldkato** [comment]
I'm going to go with option 1) as described in the above comment, i.e., giving litter its own proprietor foreign key field whose default value is automatically set to its breeding cage's proprietor (if the litter has a breeding cage). This can be accomplished by overwriting Litter's save() method, as documented here:

http://stackoverflow.com/questions/29522474/django-how-to-use-a-parents-attribute-for-a-fields-default

I've tested this out on a test database I'm running on my computer and it seems to work. Going this route offers the following features:

1) It leaves open the possibility of later transferring ownership of a litter without transferring ownership of the breeding cage it came from, 
2) it makes it easier to order litters by proprietor, and 
3) creating new litters from the breeding cage admin won't affect the proprietor of pre-existing litters from the same breeding cage (which would be a bug in those situations where a litter had been transferred to another user).

**Comment by danieldkato** [comment]
Commit aad9674 gives litter a proprietor foreign key field, and commit f267af2 makes it so that when the user creates a new litter from the breeding cage admin, the litter's proprietor is automatically set to the breeding cage's proprietor. The update takes effect when the "save" button is pressed. 

I made proprietor a non-nullable field for litter, which means that it must have a default proprietor. This means that litter must have a proprietor before the user even has a chance to press the "save" button and get the proprietor of the parent breeding cage.  For the time being, I've set the default value to the person object with id=1, which turns out to be Chris, although in the future we might want to change this to Drew.
